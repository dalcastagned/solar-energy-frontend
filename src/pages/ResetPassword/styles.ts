import styled, { keyframes } from 'styled-components';

const fadeIn = keyframes`
  from { opacity: 0; }
  to { opacity: 1; }
`;

export const Wrapper = styled.div`
  display: flex;
  height: 100vh;
  align-items: stretch;
`;

export const ContainerImage = styled.div`
  width: 50%;
  height: 100%;
  background: url('/img/login.png') center no-repeat;
  background-size: cover;

  @media screen and (max-width: 980px) {
    display: none;
  }
`;

export const ContainerLogin = styled.div`
  display: flex;
  flex-direction: column;
  width: 50%;
  justify-content: center;
  align-items: center;
  animation: ${fadeIn} 1.5s;

  @media screen and (max-width: 980px) {
    width: 100%;
    background: linear-gradient(
        135deg,
        rgba(255, 255, 255, 0.9) -10%,
        rgba(255, 255, 255, 0.9) 180%
      ),
      url('/img/login.png');
    background-size: cover;
  }
`;

export const ContainerInfo = styled.div`
  display: flex;
  flex-direction: column;
  justify-content: center;
  align-items: center;
  width: 50%;

  img {
    max-width: 28rem;

    @media screen and (max-width: 480px) {
      max-width: 18rem;
    }
  }

  h1 {
    font-size: 2.4rem;
    color: #374557;
    font-weight: 500;
    line-height: 3.2rem;
    text-align: center;
    padding: 4rem 0;

    @media screen and (max-width: 320px) {
      font-size: 2rem;
    }
  }
`;

export const Form = styled.form`
  width: 100%;
  padding: 0 2rem;
  display: flex;
  flex-direction: column;
  gap: 3rem;
  align-items: center;
`;

export const TextFieldStyles = {
  width: '100%',
  maxWidth: '51.8rem',
  label: { fontSize: '1.6rem' },
  input: { fontSize: '1.6rem' },
  '& .MuiOutlinedInput-root': { fontSize: '1.6rem' },
  '& p': { fontSize: '1.2rem' },
};

export const ButtonStyles = {
  width: '100%',
  maxWidth: '51.8rem',
  height: '5rem',
  fontSize: '2rem',
  fontWeight: '500',
};

export const AlertStyles = {
  fontSize: '1.6rem',
  width: '100%',
  maxWidth: '51.8rem',
  '& .MuiSvgIcon-root': { fontSize: '2.25rem' },
};
