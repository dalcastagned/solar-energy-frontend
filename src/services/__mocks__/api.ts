import axios from 'axios';

export type Error = {
  status?: number;
  data?: Record<string, unknown>;
  message?: string;
};

export type HttpResponse<DataType> = {
  data?: DataType;
  error?: Error;
};

const api = jest.genMockFromModule('axios') as jest.MockedFunction<
  typeof axios
>;

api.create = jest.fn(() => api);

export default api;
