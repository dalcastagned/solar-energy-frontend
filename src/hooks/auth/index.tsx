import {
  createContext,
  ReactNode,
  useCallback,
  useContext,
  useState,
} from 'react';

import jwt_decode, { JwtPayload } from 'jwt-decode';

import {
  LoginCredentials,
  UpdatePasswordCredentials,
  User,
} from '../../models/user';
import api from '../../services/api';

export type AuthState = {
  user: User;
};

export type AuthContextData = {
  user: User;
  signIn(credentials: LoginCredentials): Promise<void>;
  signOut(): void;
  updatePassword(credentials: UpdatePasswordCredentials): Promise<void>;
  signUp(credentials: LoginCredentials): Promise<void>;
};

export type AuthProviderParams = {
  children: ReactNode;
};

export const cleanStorage = (): void => {
  localStorage.removeItem('user');
};

export const handleToken = (token: string): void => {
  api.defaults.headers.common['X-JWT'] = token;
  api.defaults.headers.common.Authorization = `Bearer ${token}`;
};

const AuthContext = createContext<AuthContextData>({} as AuthContextData);

const AuthProvider = ({ children }: AuthProviderParams): JSX.Element => {
  const [data, setData] = useState<AuthState>(() => {
    const user = localStorage.getItem('user');
    const parsedUser = user ? JSON.parse(user) : null;
    const token = parsedUser?.token;

    if (token && user) {
      const tokenExpirationTime = jwt_decode<JwtPayload>(token).exp;

      const date = new Date().getTime() / 1000;

      if (tokenExpirationTime && tokenExpirationTime < date) {
        cleanStorage();

        return {} as AuthState;
      }

      handleToken(token);

      return { user: parsedUser };
    }
    return {} as AuthState;
  });

  const signIn = useCallback(async ({ email, password }: LoginCredentials) => {
    const response = await api.post<User>('/user/login', {
      email,
      password,
    });

    const { token } = response.data;
    const user = response.data;

    localStorage.setItem('user', JSON.stringify(user));

    handleToken(token);

    setData({ user });
  }, []);

  const updatePassword = useCallback(
    async ({
      user,
      currentPassword,
      newPassword,
    }: UpdatePasswordCredentials) => {
      await api.post<User>(`/user/reset-password/${user}`, {
        currentPassword,
        newPassword,
      });
    },
    [],
  );

  const signUp = useCallback(async ({ email, password }: LoginCredentials) => {
    await api.post<User>('/user/signup', {
      email,
      password,
    });
  }, []);

  const signOut = useCallback(() => {
    cleanStorage();

    setData({} as AuthState);
  }, []);

  return (
    <AuthContext.Provider
      value={{
        user: data.user,
        signIn,
        signOut,
        updatePassword,
        signUp,
      }}
    >
      {children}
    </AuthContext.Provider>
  );
};

const useAuth = (): AuthContextData => useContext(AuthContext);

export { AuthProvider, useAuth };
