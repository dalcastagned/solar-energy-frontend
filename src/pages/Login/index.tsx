import { useState } from 'react';

import EmailIcon from '@mui/icons-material/Email';
import LockIcon from '@mui/icons-material/Lock';
import { Fade, InputAdornment } from '@mui/material';
import { useFormik } from 'formik';
import { useHistory } from 'react-router-dom';
import * as Yup from 'yup';

import { useAuth } from '../../hooks/auth';
import { LoginCredentials } from '../../models/user';
import { ErrorResponse, handleErrorResponse } from '../../utils/error-response';

import * as S from './styles';

const validationSchema = Yup.object({
  email: Yup.string()
    .email('Entre com um email válido')
    .required('Email é obrigatório'),
  password: Yup.string()
    .min(6, 'Senha precisa ter no mínimo 6 caracteres')
    .max(60, 'Máximo de 60 caracteres')
    .required('Senha é obrigatória'),
});

const Login = (): JSX.Element => {
  const { signIn } = useAuth();
  const [errorMessage, setErrorMessage] = useState('');
  const [loading, setLoading] = useState(false);
  const history = useHistory();

  const handleLogin = async (credentials: LoginCredentials): Promise<void> => {
    try {
      setLoading(true);
      await signIn(credentials);
    } catch (err) {
      const errorMessageToValidate = (
        err as ErrorResponse
      ).response?.data?.toString();

      if (errorMessageToValidate?.includes('Password expired')) {
        history.push(`/reset-password`, { email: credentials.email });
      }
      setErrorMessage(handleErrorResponse(err));
      setLoading(false);
    }
  };

  const onCloseAlert = (): void => {
    setErrorMessage('');
  };

  const formik = useFormik({
    initialValues: {
      email: '',
      password: '',
    },
    validationSchema,
    onSubmit: values => {
      handleLogin(values);
    },
  });

  return (
    <S.Wrapper>
      <S.ContainerImage />
      <Fade in timeout={1000}>
        <S.ContainerLogin>
          <S.ContainerInfo>
            <img src="/img/logo.png" alt="Logo" />
            <h1>Seja bem vindo</h1>
          </S.ContainerInfo>
          <S.Form onSubmit={formik.handleSubmit}>
            {errorMessage && (
              <S.AlertStyled severity="error" onClose={onCloseAlert}>
                {errorMessage}
              </S.AlertStyled>
            )}
            <S.TextFieldStyled
              id="email"
              name="email"
              label="Email"
              InputProps={{
                placeholder: 'exemplo@exemplo.com',
                startAdornment: (
                  <InputAdornment position="start">
                    <EmailIcon />
                  </InputAdornment>
                ),
              }}
              value={formik.values.email}
              onChange={formik.handleChange}
              error={formik.touched.email && Boolean(formik.errors.email)}
              helperText={formik.touched.email && formik.errors.email}
            />
            <S.TextFieldStyled
              id="password"
              name="password"
              label="Senha"
              type="password"
              InputProps={{
                placeholder: 'Digite sua senha',
                startAdornment: (
                  <InputAdornment position="start">
                    <LockIcon />
                  </InputAdornment>
                ),
              }}
              value={formik.values.password}
              onChange={formik.handleChange}
              error={formik.touched.password && Boolean(formik.errors.password)}
              helperText={formik.touched.password && formik.errors.password}
            />
            <S.ButtonStyled
              color="primary"
              variant="contained"
              fullWidth
              type="submit"
              disabled={loading}
            >
              Entrar
            </S.ButtonStyled>
          </S.Form>
          <S.LinkStyled href="/signup">
            Ainda não possui uma conta? Cadastre-se!
          </S.LinkStyled>
        </S.ContainerLogin>
      </Fade>
    </S.Wrapper>
  );
};

export default Login;
