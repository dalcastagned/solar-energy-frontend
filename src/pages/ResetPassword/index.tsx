import { useEffect, useState } from 'react';

import LockIcon from '@mui/icons-material/Lock';
import { CircularProgress, Fade, InputAdornment } from '@mui/material';
import { useFormik } from 'formik';
import { useHistory, useLocation } from 'react-router-dom';
import * as Yup from 'yup';

import { useAuth } from '../../hooks/auth';
import { handleErrorResponse } from '../../utils/error-response';

import * as S from './styles';

const validationSchema = Yup.object().shape({
  currentPassword: Yup.string()
    .min(6, 'Senha precisa ter no mínimo 6 caracteres')
    .required('A senha atual é obrigatória.')
    .max(60, 'Máximo de 60 caracteres'),
  newPassword: Yup.string()
    .min(6, 'Senha precisa ter no mínimo 6 caracteres')
    .required('A senha nova é obrigatória.')
    .max(60, 'Máximo de 60 caracteres'),
  confirmNewPassword: Yup.string()
    .min(6, 'Senha precisa ter no mínimo 6 caracteres')
    .required('A confirmação de senha é obrigatória.')
    .oneOf([Yup.ref('newPassword'), null], 'Senhas não correspondem')
    .max(60, 'Máximo de 60 caracteres'),
});

const ResetPassword = (): JSX.Element => {
  const { updatePassword } = useAuth();
  const [errorMessage, setErrorMessage] = useState('');
  const [loading, setLoading] = useState(false);
  const history = useHistory();
  const location = useLocation();

  useEffect(() => {
    if (!location.state?.email) {
      history.push('/');
    }
  }, [history, location.state?.email]);

  const handleUpdatePassword = async (
    currentPassword: string,
    newPassword: string,
  ): Promise<void> => {
    try {
      setLoading(true);
      await updatePassword({
        user: location.state?.email,
        currentPassword,
        newPassword,
      });
      history.push('/');
    } catch (err) {
      setErrorMessage(handleErrorResponse(err, location.pathname));
      setLoading(false);
    }
  };

  const onCloseAlert = (): void => {
    setErrorMessage('');
  };

  const formik = useFormik({
    initialValues: {
      currentPassword: '',
      newPassword: '',
      confirmNewPassword: '',
    },
    validationSchema,
    onSubmit: values => {
      handleUpdatePassword(values.currentPassword, values.newPassword);
    },
  });

  return (
    <S.Wrapper>
      <S.ContainerImage />
      <Fade in timeout={1000}>
        <S.ContainerResetPassword>
          <S.ContainerInfo>
            <img src="/img/logo.png" alt="Logo" />
            <h1>Senha expirada. Atualize para continuar!</h1>
          </S.ContainerInfo>
          <S.Form onSubmit={formik.handleSubmit}>
            {errorMessage && (
              <S.AlertStyled severity="error" onClose={onCloseAlert}>
                {errorMessage}
              </S.AlertStyled>
            )}
            <S.TextFieldStyled
              id="currentPassword"
              name="currentPassword"
              label="Senha atual"
              type="password"
              InputProps={{
                placeholder: 'Digite sua senha atual',
                startAdornment: (
                  <InputAdornment position="start">
                    <LockIcon />
                  </InputAdornment>
                ),
              }}
              value={formik.values.currentPassword}
              onChange={formik.handleChange}
              error={
                formik.touched.currentPassword &&
                Boolean(formik.errors.currentPassword)
              }
              helperText={
                formik.touched.currentPassword && formik.errors.currentPassword
              }
            />
            <S.TextFieldStyled
              id="newPassword"
              name="newPassword"
              label="Nova senha"
              type="password"
              InputProps={{
                placeholder: 'Digite sua nova senha',
                startAdornment: (
                  <InputAdornment position="start">
                    <LockIcon />
                  </InputAdornment>
                ),
              }}
              value={formik.values.newPassword}
              onChange={formik.handleChange}
              error={
                formik.touched.newPassword && Boolean(formik.errors.newPassword)
              }
              helperText={
                formik.touched.newPassword && formik.errors.newPassword
              }
            />
            <S.TextFieldStyled
              id="confirmNewPassword"
              name="confirmNewPassword"
              label="Confirmar nova senha"
              type="password"
              InputProps={{
                placeholder: 'Confirme sua nova senha',
                startAdornment: (
                  <InputAdornment position="start">
                    <LockIcon />
                  </InputAdornment>
                ),
              }}
              value={formik.values.confirmNewPassword}
              onChange={formik.handleChange}
              error={
                formik.touched.confirmNewPassword &&
                Boolean(formik.errors.confirmNewPassword)
              }
              helperText={
                formik.touched.confirmNewPassword &&
                formik.errors.confirmNewPassword
              }
            />
            <S.ButtonStyled
              color="primary"
              variant="contained"
              fullWidth
              type="submit"
            >
              {loading ? <CircularProgress color="inherit" /> : 'Alterar senha'}
            </S.ButtonStyled>
          </S.Form>
        </S.ContainerResetPassword>
      </Fade>
    </S.Wrapper>
  );
};

export default ResetPassword;
