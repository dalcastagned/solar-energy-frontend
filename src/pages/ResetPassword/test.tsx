import {
  act,
  screen,
  waitFor,
  waitForElementToBeRemoved,
} from '@testing-library/react';
import userEvent from '@testing-library/user-event';
import { createMemoryHistory } from 'history';
import { Router } from 'react-router-dom';

import ResetPassword from '.';
import Login from '.';
import { AuthProvider } from '../../hooks/auth';
import Route from '../../routes/Route';
import api from '../../services/api';
import { renderWithRoute } from '../../utils/tests/helpers';

describe('<ResetPassword />', () => {
  it('should render the main', () => {
    const { container } = renderWithRoute(<ResetPassword />);

    expect(screen.getByText('Alterar senha')).toBeInTheDocument();
    expect(screen.getByLabelText('Senha atual')).toBeInTheDocument();
    expect(screen.getByLabelText('Nova senha')).toBeInTheDocument();
    expect(screen.getByLabelText('Confirmar nova senha')).toBeInTheDocument();
    expect(container.firstChild).toMatchSnapshot();
  });

  it('should render the validation messages when update passowd without currentPassword, newPassord and confirmNewPassword', async () => {
    renderWithRoute(<ResetPassword />);

    const updatePasswordButton = screen.getByRole('button', {
      name: 'Alterar senha',
    });

    userEvent.click(updatePasswordButton);

    await waitFor(() => {
      expect(
        screen.getByText('A senha atual é obrigatória.'),
      ).toBeInTheDocument();
      expect(
        screen.getByText('A senha nova é obrigatória.'),
      ).toBeInTheDocument();
      expect(
        screen.getByText('A confirmação de senha é obrigatória.'),
      ).toBeInTheDocument();
    });
  });

  it('must render validation messages when currentPassword, newPassord and confirmNewPassword contain more than 60 characters', async () => {
    renderWithRoute(<Login />);

    const currentPasswordInput = screen.getByLabelText('Senha atual');
    userEvent.type(currentPasswordInput, Array(61).fill('a').join(''));

    const newPasswordInput = screen.getByLabelText('Nova senha');
    userEvent.type(newPasswordInput, Array(61).fill('a').join(''));

    const confirmNewPasswordInput = screen.getByLabelText(
      'Confirmar nova senha',
    );
    userEvent.type(confirmNewPasswordInput, Array(61).fill('a').join(''));

    const updatePasswordButton = screen.getByRole('button', {
      name: 'Alterar senha',
    });

    userEvent.click(updatePasswordButton);

    await waitFor(() => {
      const [
        currentPasswordValidation,
        newPasswordValidation,
        confirmNewPasswordValidation,
      ] = screen.getAllByText('Máximo de 60 caracteres');
      expect(currentPasswordValidation).toBeInTheDocument();
      expect(newPasswordValidation).toBeInTheDocument();
      expect(confirmNewPasswordValidation).toBeInTheDocument();
    });
  });

  it('should render alert when current password is not correct', async () => {
    jest.spyOn(api, 'post').mockRejectedValue({ response: { status: 400 } });

    const history = createMemoryHistory();
    history.push(`/reset-password`, { email: 'email@email.com' });

    renderWithRoute(
      <AuthProvider>
        <Router history={history}>
          <ResetPassword />
        </Router>
      </AuthProvider>,
    );

    const currentPasswordInput = screen.getByLabelText('Senha atual');
    userEvent.type(currentPasswordInput, 'senha1');

    const newPasswordInput = screen.getByLabelText('Nova senha');
    userEvent.type(newPasswordInput, 'senhanova');

    const confirmNewPasswordInput = screen.getByLabelText(
      'Confirmar nova senha',
    );
    userEvent.type(confirmNewPasswordInput, 'senhanova');

    const updatePasswordButton = screen.getByRole('button', {
      name: 'Alterar senha',
    });

    userEvent.click(updatePasswordButton);

    await waitFor(() => {
      expect(screen.getByText('Senha atual incorreta.')).toBeInTheDocument();
    });
  });

  it('should render alert when server is down', async () => {
    jest.spyOn(api, 'post').mockRejectedValue({ code: 'ECONNABORTED' });

    const history = createMemoryHistory();
    history.push(`/reset-password`, { email: 'email@email.com' });

    renderWithRoute(
      <AuthProvider>
        <Router history={history}>
          <ResetPassword />
        </Router>
      </AuthProvider>,
    );

    const currentPasswordInput = screen.getByLabelText('Senha atual');
    userEvent.type(currentPasswordInput, 'senha1');

    const newPasswordInput = screen.getByLabelText('Nova senha');
    userEvent.type(newPasswordInput, 'senhanova');

    const confirmNewPasswordInput = screen.getByLabelText(
      'Confirmar nova senha',
    );
    userEvent.type(confirmNewPasswordInput, 'senhanova');

    const updatePasswordButton = screen.getByRole('button', {
      name: 'Alterar senha',
    });

    userEvent.click(updatePasswordButton);

    await waitFor(async () => {
      expect(
        screen.getByText(
          'O servidor parece estar indisponível. Verifique sua conexão.',
        ),
      ).toBeInTheDocument();
    });
  });

  it('should render alert with server error message', async () => {
    jest.spyOn(api, 'post').mockRejectedValue({ response: { status: 500 } });

    const history = createMemoryHistory();
    history.push(`/reset-password`, { email: 'email@email.com' });

    renderWithRoute(
      <AuthProvider>
        <Router history={history}>
          <ResetPassword />
        </Router>
      </AuthProvider>,
    );

    const currentPasswordInput = screen.getByLabelText('Senha atual');
    userEvent.type(currentPasswordInput, 'senha1');

    const newPasswordInput = screen.getByLabelText('Nova senha');
    userEvent.type(newPasswordInput, 'senhanova');

    const confirmNewPasswordInput = screen.getByLabelText(
      'Confirmar nova senha',
    );
    userEvent.type(confirmNewPasswordInput, 'senhanova');

    const updatePasswordButton = screen.getByRole('button', {
      name: 'Alterar senha',
    });

    userEvent.click(updatePasswordButton);

    await waitFor(() => {
      expect(
        screen.getByText('Ocorreu um erro no servidor.'),
      ).toBeInTheDocument();
    });
  });

  it('should close alert when clicked on closeButton', async () => {
    jest.spyOn(api, 'post').mockRejectedValue({ response: undefined });

    const history = createMemoryHistory();
    history.push(`/reset-password`, { email: 'email@email.com' });

    renderWithRoute(
      <AuthProvider>
        <Router history={history}>
          <ResetPassword />
        </Router>
      </AuthProvider>,
    );

    const currentPasswordInput = screen.getByLabelText('Senha atual');
    userEvent.type(currentPasswordInput, 'senha1');

    const newPasswordInput = screen.getByLabelText('Nova senha');
    userEvent.type(newPasswordInput, 'senhanova');

    const confirmNewPasswordInput = screen.getByLabelText(
      'Confirmar nova senha',
    );
    userEvent.type(confirmNewPasswordInput, 'senhanova');

    const updatePasswordButton = screen.getByRole('button', {
      name: 'Alterar senha',
    });

    userEvent.click(updatePasswordButton);

    await waitFor(() => {
      const closeAlert = screen.getByLabelText('Close');
      userEvent.click(closeAlert);

      expect(screen.queryByRole('alert')).not.toBeInTheDocument();
    });
  });

  it('should render alert when password confirmation is different from new password', async () => {
    renderWithRoute(
      <AuthProvider>
        <ResetPassword />
      </AuthProvider>,
    );

    const currentPasswordInput = screen.getByLabelText('Senha atual');
    userEvent.type(currentPasswordInput, 'senha1');

    const newPasswordInput = screen.getByLabelText('Nova senha');
    userEvent.type(newPasswordInput, 'senhanova');

    const confirmNewPasswordInput = screen.getByLabelText(
      'Confirmar nova senha',
    );
    userEvent.type(confirmNewPasswordInput, 'senhanov');

    const updatePasswordButton = screen.getByRole('button', {
      name: 'Alterar senha',
    });

    userEvent.click(updatePasswordButton);

    await waitFor(() => {
      expect(screen.getByText('Senhas não correspondem')).toBeInTheDocument();
    });
  });

  it('should go to login screen when password change is successful', async () => {
    jest.spyOn(api, 'post').mockImplementation(jest.fn());

    const history = createMemoryHistory();
    history.push(`/reset-password`, { email: 'email@email.com' });

    const loginPage = (): JSX.Element => <h1>Login</h1>;

    renderWithRoute(
      <AuthProvider>
        <Router history={history}>
          <ResetPassword />
          <Route Component={loginPage} path="/" />
        </Router>
      </AuthProvider>,
    );

    const currentPasswordInput = screen.getByLabelText('Senha atual');
    userEvent.type(currentPasswordInput, 'senha1');

    const newPasswordInput = screen.getByLabelText('Nova senha');
    userEvent.type(newPasswordInput, 'senhanova');

    const confirmNewPasswordInput = screen.getByLabelText(
      'Confirmar nova senha',
    );
    userEvent.type(confirmNewPasswordInput, 'senhanova');

    const updatePasswordButton = screen.getByRole('button', {
      name: 'Alterar senha',
    });

    await act(async () => {
      userEvent.click(updatePasswordButton);

      await waitFor(() => {
        expect(screen.getByText('Login')).toBeInTheDocument();
      });
    });
  });
});
