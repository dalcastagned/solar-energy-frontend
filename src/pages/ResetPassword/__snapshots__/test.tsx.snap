// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<ResetPassword /> should render the main 1`] = `
.c0 {
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  height: 100vh;
  -webkit-align-items: stretch;
  -webkit-box-align: stretch;
  -ms-flex-align: stretch;
  align-items: stretch;
}

.c1 {
  width: 50%;
  height: 100%;
  background: url('/img/login.png') center no-repeat;
  background-size: cover;
}

.c2 {
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-flex-direction: column;
  -ms-flex-direction: column;
  flex-direction: column;
  width: 50%;
  -webkit-box-pack: center;
  -webkit-justify-content: center;
  -ms-flex-pack: center;
  justify-content: center;
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
  -webkit-animation: hUniol 1.5s;
  animation: hUniol 1.5s;
}

.c3 {
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-flex-direction: column;
  -ms-flex-direction: column;
  flex-direction: column;
  -webkit-box-pack: center;
  -webkit-justify-content: center;
  -ms-flex-pack: center;
  justify-content: center;
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
  width: 50%;
}

.c3 img {
  max-width: 28rem;
}

.c3 h1 {
  font-size: 2.4rem;
  color: #374557;
  font-weight: 500;
  line-height: 3.2rem;
  text-align: center;
  padding: 4rem 0;
}

.c4 {
  width: 100%;
  padding: 0 2rem;
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-flex-direction: column;
  -ms-flex-direction: column;
  flex-direction: column;
  gap: 3rem;
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
}

@media screen and (max-width:980px) {
  .c1 {
    display: none;
  }
}

@media screen and (max-width:980px) {
  .c2 {
    width: 100%;
    background: linear-gradient( 135deg, rgba(255,255,255,0.9) -10%, rgba(255,255,255,0.9) 180% ), url('/img/login.png');
    background-size: cover;
  }
}

@media screen and (max-width:480px) {
  .c3 img {
    max-width: 18rem;
  }
}

@media screen and (max-width:320px) {
  .c3 h1 {
    font-size: 2rem;
  }
}

<div
  class="c0"
>
  <div
    class="c1"
  />
  <div
    class="c2"
  >
    <div
      class="c3"
    >
      <img
        alt="Logo"
        src="/img/logo.png"
      />
      <h1>
        Senha expirada. Atualize para continuar!
      </h1>
    </div>
    <form
      class="c4"
    >
      <div
        class="MuiFormControl-root MuiTextField-root css-1iich4g-MuiFormControl-root-MuiTextField-root"
      >
        <label
          class="MuiInputLabel-root MuiInputLabel-formControl MuiInputLabel-animated MuiInputLabel-shrink MuiInputLabel-outlined MuiFormLabel-root MuiFormLabel-colorPrimary css-1sumxir-MuiFormLabel-root-MuiInputLabel-root"
          data-shrink="true"
          for="currentPassword"
          id="currentPassword-label"
        >
          Senha atual
        </label>
        <div
          class="MuiOutlinedInput-root MuiInputBase-root MuiInputBase-colorPrimary MuiInputBase-formControl MuiInputBase-adornedStart css-1q6at85-MuiInputBase-root-MuiOutlinedInput-root"
        >
          <div
            class="MuiInputAdornment-root MuiInputAdornment-positionStart MuiInputAdornment-outlined MuiInputAdornment-sizeMedium css-ittuaa-MuiInputAdornment-root"
          >
            <span
              class="notranslate"
            >
              ​
            </span>
            <svg
              aria-hidden="true"
              class="MuiSvgIcon-root MuiSvgIcon-fontSizeMedium css-i4bv87-MuiSvgIcon-root"
              data-testid="LockIcon"
              focusable="false"
              viewBox="0 0 24 24"
            >
              <path
                d="M18 8h-1V6c0-2.76-2.24-5-5-5S7 3.24 7 6v2H6c-1.1 0-2 .9-2 2v10c0 1.1.9 2 2 2h12c1.1 0 2-.9 2-2V10c0-1.1-.9-2-2-2zm-6 9c-1.1 0-2-.9-2-2s.9-2 2-2 2 .9 2 2-.9 2-2 2zm3.1-9H8.9V6c0-1.71 1.39-3.1 3.1-3.1 1.71 0 3.1 1.39 3.1 3.1v2z"
              />
            </svg>
          </div>
          <input
            aria-invalid="false"
            class="MuiOutlinedInput-input MuiInputBase-input MuiInputBase-inputAdornedStart css-1o9s3wi-MuiInputBase-input-MuiOutlinedInput-input"
            id="currentPassword"
            name="currentPassword"
            placeholder="Digite sua senha atual"
            type="password"
            value=""
          />
          <fieldset
            aria-hidden="true"
            class="MuiOutlinedInput-notchedOutline css-1d3z3hw-MuiOutlinedInput-notchedOutline"
          >
            <legend
              class="css-1in441m"
            >
              <span>
                Senha atual
              </span>
            </legend>
          </fieldset>
        </div>
      </div>
      <div
        class="MuiFormControl-root MuiTextField-root css-1iich4g-MuiFormControl-root-MuiTextField-root"
      >
        <label
          class="MuiInputLabel-root MuiInputLabel-formControl MuiInputLabel-animated MuiInputLabel-shrink MuiInputLabel-outlined MuiFormLabel-root MuiFormLabel-colorPrimary css-1sumxir-MuiFormLabel-root-MuiInputLabel-root"
          data-shrink="true"
          for="newPassword"
          id="newPassword-label"
        >
          Nova senha
        </label>
        <div
          class="MuiOutlinedInput-root MuiInputBase-root MuiInputBase-colorPrimary MuiInputBase-formControl MuiInputBase-adornedStart css-1q6at85-MuiInputBase-root-MuiOutlinedInput-root"
        >
          <div
            class="MuiInputAdornment-root MuiInputAdornment-positionStart MuiInputAdornment-outlined MuiInputAdornment-sizeMedium css-ittuaa-MuiInputAdornment-root"
          >
            <span
              class="notranslate"
            >
              ​
            </span>
            <svg
              aria-hidden="true"
              class="MuiSvgIcon-root MuiSvgIcon-fontSizeMedium css-i4bv87-MuiSvgIcon-root"
              data-testid="LockIcon"
              focusable="false"
              viewBox="0 0 24 24"
            >
              <path
                d="M18 8h-1V6c0-2.76-2.24-5-5-5S7 3.24 7 6v2H6c-1.1 0-2 .9-2 2v10c0 1.1.9 2 2 2h12c1.1 0 2-.9 2-2V10c0-1.1-.9-2-2-2zm-6 9c-1.1 0-2-.9-2-2s.9-2 2-2 2 .9 2 2-.9 2-2 2zm3.1-9H8.9V6c0-1.71 1.39-3.1 3.1-3.1 1.71 0 3.1 1.39 3.1 3.1v2z"
              />
            </svg>
          </div>
          <input
            aria-invalid="false"
            class="MuiOutlinedInput-input MuiInputBase-input MuiInputBase-inputAdornedStart css-1o9s3wi-MuiInputBase-input-MuiOutlinedInput-input"
            id="newPassword"
            name="newPassword"
            placeholder="Digite sua nova senha"
            type="password"
            value=""
          />
          <fieldset
            aria-hidden="true"
            class="MuiOutlinedInput-notchedOutline css-1d3z3hw-MuiOutlinedInput-notchedOutline"
          >
            <legend
              class="css-1in441m"
            >
              <span>
                Nova senha
              </span>
            </legend>
          </fieldset>
        </div>
      </div>
      <div
        class="MuiFormControl-root MuiTextField-root css-1iich4g-MuiFormControl-root-MuiTextField-root"
      >
        <label
          class="MuiInputLabel-root MuiInputLabel-formControl MuiInputLabel-animated MuiInputLabel-shrink MuiInputLabel-outlined MuiFormLabel-root MuiFormLabel-colorPrimary css-1sumxir-MuiFormLabel-root-MuiInputLabel-root"
          data-shrink="true"
          for="confirmNewPassword"
          id="confirmNewPassword-label"
        >
          Confirmar nova senha
        </label>
        <div
          class="MuiOutlinedInput-root MuiInputBase-root MuiInputBase-colorPrimary MuiInputBase-formControl MuiInputBase-adornedStart css-1q6at85-MuiInputBase-root-MuiOutlinedInput-root"
        >
          <div
            class="MuiInputAdornment-root MuiInputAdornment-positionStart MuiInputAdornment-outlined MuiInputAdornment-sizeMedium css-ittuaa-MuiInputAdornment-root"
          >
            <span
              class="notranslate"
            >
              ​
            </span>
            <svg
              aria-hidden="true"
              class="MuiSvgIcon-root MuiSvgIcon-fontSizeMedium css-i4bv87-MuiSvgIcon-root"
              data-testid="LockIcon"
              focusable="false"
              viewBox="0 0 24 24"
            >
              <path
                d="M18 8h-1V6c0-2.76-2.24-5-5-5S7 3.24 7 6v2H6c-1.1 0-2 .9-2 2v10c0 1.1.9 2 2 2h12c1.1 0 2-.9 2-2V10c0-1.1-.9-2-2-2zm-6 9c-1.1 0-2-.9-2-2s.9-2 2-2 2 .9 2 2-.9 2-2 2zm3.1-9H8.9V6c0-1.71 1.39-3.1 3.1-3.1 1.71 0 3.1 1.39 3.1 3.1v2z"
              />
            </svg>
          </div>
          <input
            aria-invalid="false"
            class="MuiOutlinedInput-input MuiInputBase-input MuiInputBase-inputAdornedStart css-1o9s3wi-MuiInputBase-input-MuiOutlinedInput-input"
            id="confirmNewPassword"
            name="confirmNewPassword"
            placeholder="Confirme sua nova senha"
            type="password"
            value=""
          />
          <fieldset
            aria-hidden="true"
            class="MuiOutlinedInput-notchedOutline css-1d3z3hw-MuiOutlinedInput-notchedOutline"
          >
            <legend
              class="css-1in441m"
            >
              <span>
                Confirmar nova senha
              </span>
            </legend>
          </fieldset>
        </div>
      </div>
      <button
        class="MuiButton-root MuiButton-contained MuiButton-containedPrimary MuiButton-sizeMedium MuiButton-containedSizeMedium MuiButton-fullWidth MuiButtonBase-root  css-vyxd6w-MuiButtonBase-root-MuiButton-root"
        tabindex="0"
        type="submit"
      >
        Alterar senha
        <span
          class="MuiTouchRipple-root css-8je8zh-MuiTouchRipple-root"
        />
      </button>
    </form>
  </div>
</div>
`;
